### PR TITLE
CI/TST: Supply dtype to coo_matrix until scipy is fixed

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

Fixes CI for Windows 3.7 np1.6; regression in scipy 1.6.1 (https://github.com/scipy/scipy/issues/13585)

Alternative would be to pin build, wasn't sure which is prefered over the other. Will open tracking issue once the method is decided.

### CODE REVIEW

File: pandas/tests/arrays/sparse/test_array.py [hunk 1]
```diff
@@ -1185,7 +1185,9 @@ def test_from_coo(self):
         row = [0, 3, 1, 0]
         col = [0, 3, 1, 2]
         data = [4, 5, 7, 9]
-        sp_array = scipy.sparse.coo_matrix((data, (row, col)))
+        # TODO: Remove dtype when scipy is fixed
+        # https://github.com/scipy/scipy/issues/13585
+        sp_array = scipy.sparse.coo_matrix((data, (row, col)), dtype="int64")
```
Comment (jreback) [inline pandas/tests/arrays/sparse/test_array.py:1190]: maybe pass intp here?

**Comment by jreback** [comment]
going to backport as well, ping on green.

**Comment by rhshadrach** [comment]
Windows 3.7 np1.6, failures look unrelated. Running again.

**Comment by rhshadrach** [comment]
@jreback - greenish

Failing test is `TestTimedeltas.test_to_timedelta_oob_non_nano`, I'm seeing this fail on the most recent commit to master as well.

**Comment by jreback** [comment]
> @jreback - greenish
> 
> Failing test is `TestTimedeltas.test_to_timedelta_oob_non_nano`, I'm seeing this fail on the most recent commit to master as well.

oh that's prob something we just merged cc @jbrockmendel

**Comment by jreback** [comment]
thanks @rhshadrach

**Comment by jreback** [comment]
@meeseeksdev backport 1.2.x

**Comment by jbrockmendel** [comment]
> test_to_timedelta_oob_non_nano

will take a look

**Pull request merged**